### DISCUSSION TITLE
Tool For Opaque Predicates Detection

### DISCUSSION THREAD

**Discussion by yellowbyte** [Vector35/binaryninja-api#2576] [Ideas] [answered]
I made a headless binja script that detects opaque predicates similar to OpaquePredicatePatcher in that it uses the value set to detect an opaque predicate. However, it differs in that it will evaluate each value in the value set against the predicate to determine if it is opaque or not, instead of checking if the cardinality of the value set is one.

I'm thinking of making it into a plugin. Should I make a PR to OpaquePredicatePatcher or should I submit it as a new plugin? I wanted to ask first before just making the PR since it will change the way OpaquePredicatePatcher works.

**Comment by psifertex** [comment, marked as answer]
Sorry for the late reply -- we've been remiss in making sure we tracked discussions lately. 

The short explanation is we're more than happy to have a PR for the OpaquePredicate patcher -- however I'd highly recommend checking out the new Workflows APIs on dev (if you have a commercial license -- we don't have a specific release date on when it will be in personal unfortunately) as it can allow you to actually replace things in the analysis without relying on native "patch" functionality (this could potentially allow much more powerful transformations).

**Reply by yellowbyte** [reply]
That sounds amazing. I will check it out!
